processing/src/main/java/org/apache/druid/segment/incremental/OnheapIncrementalIndex.java

```diff
@@ -143,6 +144,9 @@ private void initializeProjections(IncrementalIndexSchema incrementalIndexSchema
       // initialize them all with 0 rows
       AggregateProjectionMetadata.Schema schema = projectionSpec.toMetadataSchema();
       aggregateProjections.add(new AggregateProjectionMetadata(schema, 0));
+      if (projections.containsKey(projectionSpec.getName())) {
+        throw new IAE("Found duplicate projection name[%s].", projectionSpec.getName());
```

Comment:
Style:nit this should be a druid exception. This might be a user error so we might need to add a better exception message saying remove the projection and try to resubmit the ingestion job.